### PR TITLE
#3030 Introduce performance test for FtpSource

### PR DESCRIPTION
**Pull request by chia7712**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3030)
- [x] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-it/src/test/scala/com/island/ohara/it/performance/BasicTestPerformance4Ftp.scala [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+import com.island.ohara.common.util.{CommonUtils, Releasable}
+import com.island.ohara.testing.service.FtpServer
+import org.junit.After
+import spray.json.{JsNumber, JsString, JsValue}
+abstract class BasicTestPerformance4Ftp extends BasicTestPerformance {
+  /**
+    * TODO: we should use individual ftp server rather than embedded ftp
+    * (fix by https://github.com/oharastream/ohara/issues/3030)
+    */
+  private[this] val ftpServer = FtpServer
```
Comment (jackyoh) [inline ohara-it/src/test/scala/com/island/ohara/it/performance/BasicTestPerformance4Ftp.scala]: 這邊之後整 container, 我會再來修改程式碼

Comment (chia7712) [reply ohara-it/src/test/scala/com/island/ohara/it/performance/BasicTestPerformance4Ftp.scala]: U are right :)

**Comment by chia7712** [comment]
retry collieIT

File: ohara-it/src/test/scala/com/island/ohara/it/performance/BasicTestPerformance4Ftp.scala [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+      .password(ftpServer.password())
+      .build
+
+  protected def setupInputData(): (String, Long, Long) = {
+    val pool        = Executors.newFixedThreadPool(numberOfProducerThread)
+    val closed      = new AtomicBoolean(false)
+    val count       = new LongAdder()
+    val sizeInBytes = new LongAdder()
+
+    val client = ftpClient()
```
Comment (jackyoh) [inline ohara-it/src/test/scala/com/island/ohara/it/performance/BasicTestPerformance4Ftp.scala]: 75 行的 client 可以跟 83 行的 client 共用嗎？

Comment (chia7712) [reply ohara-it/src/test/scala/com/island/ohara/it/performance/BasicTestPerformance4Ftp.scala]: FtpClient is NOT thread-safe so each thread MUST have its client instance :(

Comment (jackyoh) [reply ohara-it/src/test/scala/com/island/ohara/it/performance/BasicTestPerformance4Ftp.scala]: OK, 了解。

**Pull request merged**